src/Core/Injector/Injector.php

```diff
@@ -524,16 +524,28 @@ public function convertServiceProperty($value)
         }
 
         // Evaluate constants surrounded by back ticks
-        if (preg_match('/^`(?<name>[^`]+)`$/', $value ?? '', $matches)) {
-            $envValue = Environment::getEnv($matches['name']);
+        $hasBacticks = false;
+        $allMissing = true;
+        $rx = '/`(?<name>[^`]+)`/';
+        while (preg_match($rx, $value ?? '', $matches)) {
+            $hasBacticks = true;
+            $name = $matches['name'];
+            $envValue = Environment::getEnv($name);
+            $val = '';
             if ($envValue !== false) {
-                $value = $envValue;
-            } elseif (defined($matches['name'] ?? '')) {
-                $value = constant($matches['name'] ?? '');
-            } else {
-                $value = null;
+                $val = $envValue;
+            } elseif (defined($name)) {
+                $val = constant($name);
+            }
+            $value = str_replace("`$name`", $val, $value);
```

Comment:
If there is no value for `$val` here is there potential here that we could be breaking someone's config? e.g. where previously something like ``'some value `with a tick` here'`` would have just been used as is, now it will be turned into `some value  here`.

Reply:
I don't think we need to consider this edge case, because in Silverstripe backticks are used to denote string substituting in an environment variable or constant

The proper way to deal with this would be to have specific logic for escaping backticks to prevent string substitution, though I be surprised if there's a single example of someone doing this in the wild, so I don't think we need to solve this right now

Reply:
Doing a regex search in vscode for a sink project
`['"][^`]+``
`**/*.yml` (includes ALL yml in the vendor folder)

The only matches are in `vendor/silverstripe/framework/.upgrade.yml` e.g.
```
      message: 'Session can not be accessed statically and `get_all()` is now called `getAll()'
    'SilverStripe\Control\Session::clear_all()':
```

So, perhaps there is a slightly concern we'd chop a few strings?  If this is a concern then we'd need to reconsider this whole thing

Reply:
> in Silverstripe backticks are used to denote string substituting in an environment variable or constant

This is only true (before this PR) of a value that entirely consists of an environment variable or constant - the regex used (``^`(?<name>[^`]+)`$``) means that only if there was a backtick at the beginning and the end of the value would it be used in this way. We're changing that with this PR.

> I be surprised if there's a single example of someone doing this in the wild

I wouldn't be surprised at all to see someone throwing backticks in their config for any number of reasons (javascript template literal that they'll later inject somewhere in a templated javascript file or `Requirements::customScript()` call, surrounding table or column names in something intended for use in a direct SQL query, or for some arbitrary custom purpose)
I don't think we can discount this outright.

Reply:
We can avoid this by _not_ replacing these with an empty value in situations where the value isn't entirely an env var or constant. 

e.g.
```yml
SomeClass:
  some_config: '`THIS_WILL_BE_EMPTY`'
  some_config: '`this will be empty`'
  some_config: 'the following `will not` be empty'
```

Reply:
I've updated the logic so that `$value` must start and end with a backtick, though you can have multiple things that get substituted within `$value

This should make this change backwards compatible